engine/src/main/java/org/hibernate/search/engine/spi/SearchMappingHelper.java

```diff
@@ -81,7 +82,7 @@ private static SearchMapping getNewInstanceOfSearchMapping(Class<?> clazz, Metho
 		SearchMapping mapping;
 		try {
 			LOG.debugf( "invoking factory method [ %s.%s ] to get search mapping instance", clazz.getName(), method.getName() );
-			Object instance = clazz.newInstance();
+			final Object instance = ClassLoaderHelper.untypedInstanceFromClass( clazz, "SearchMapping provider" );
```

Comment:
Real question: what does this `final` bring us?

Reply:
readability

Reply:
Yeah, it's mostly readability. If Java used constants instead of variables by default, there would be reasons to use some `mutable` or `var` keyword from time to time, but since it's the other way around, it's just easier to leave everything mutable.

I don't think the compiler is able to optimize things more when the `final` keyword is used on a local variable, since even without it the compiler is able to detect "effectively final" variables. The "effectively final" concept is used for lambdas: it's what allows you to refer to a variable out of the scope of the lambda, on the condition the variable is never re-affected. 

Previously (Java 7 and below) you had to mark a variable as `final` to use it in an anonymous class, but with Java 8 and this "effectively final" concept, I'm not sure it's necessary any longer. I should try it out some time.